lib/rules/await-async-events.ts

```diff
@@ -0,0 +1,227 @@
+import { ASTUtils, TSESLint, TSESTree } from '@typescript-eslint/utils';
+
+import { createTestingLibraryRule } from '../create-testing-library-rule';
+import {
+	findClosestCallExpressionNode,
+	findClosestFunctionExpressionNode,
+	getFunctionName,
+	getInnermostReturningFunction,
+	getVariableReferences,
+	isMemberExpression,
+	isPromiseHandled,
+} from '../node-utils';
+import { EVENTS_SIMULATORS } from '../utils';
+
+export const RULE_NAME = 'await-async-events';
+export type MessageIds = 'awaitAsyncEvent' | 'awaitAsyncEventWrapper';
+const FIRE_EVENT_NAME = 'fireEvent';
+const USER_EVENT_NAME = 'userEvent';
+type EventModules = (typeof EVENTS_SIMULATORS)[number];
+export type Options = [
+	{
+		eventModule: EventModules | EventModules[];
+	}
+];
+
+export default createTestingLibraryRule<Options, MessageIds>({
+	name: RULE_NAME,
+	meta: {
+		type: 'problem',
+		docs: {
+			description: 'Enforce promises from async event methods are handled',
+			recommendedConfig: {
+				dom: ['error', { eventModule: 'userEvent' }],
+				angular: ['error', { eventModule: 'userEvent' }],
+				react: ['error', { eventModule: 'userEvent' }],
```

Comment:
Should we use an array here as well for consistency?

```suggestion
				dom: ['error', { eventModule: ['userEvent'] }],
				angular: ['error', { eventModule: ['userEvent'] }],
				react: ['error', { eventModule: ['userEvent'] }],
```